jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/java/corrections/proposal/ModifyAnnotationProposal.java

```diff
@@ -263,10 +221,137 @@ protected ASTRewrite getRewrite() throws CoreException {
                     newRewrite = TypeDeclaration.MODIFIERS2_PROPERTY;
                 }
 
-                rewrite.getListRewrite(declNode, newRewrite).insertFirst(marker, null);
+                // add new annotation proposal to the rewrite text edit
+                rewrite.getListRewrite(declNode, newRewrite).insertFirst(newAnnotationToWrite, null);
             }
             return rewrite;
         }
         return null;
     }
+
+    private SingleMemberAnnotation processSingleMemberAnnotation(AST ast, ImportRewrite imports,
+                                                                 ImportRewriteContext importRWCtx, String[] annotations,
+                                                                 SingleMemberAnnotation annotationToProcess) {
+
+        // A SingleMemberAnnotation is an annotation that contains within it a single
+        // value (with no name associated with it)
+        // That value may be a list of additional annotations, IE:
+        //
+        // @Resources ({ @Resource(name = "aaa"), @Resource(type = Object.class) })
+        //
+        // This method will process the annotations in the list, one by one, for quick
+        // fix actions on any applicable sub annotations
+
+        SingleMemberAnnotation newSingleMemberAnnotation = ast.newSingleMemberAnnotation();
+        ArrayInitializer newAIInstance = (ArrayInitializer) ast.createInstance(ArrayInitializer.class);
+
+        newSingleMemberAnnotation.setTypeName(
+                                              ast.newName(imports.addImport(annotationToProcess.getTypeName().toString(), importRWCtx)));
+        List<NormalAnnotation> newCompositeAnnotationContents = newAIInstance.expressions();
```

Comment:
It looks like you populate this: newCompositeAnnotationContents, but it is never used.